### ISSUE TITLE
is the application currently considered reliable?

### ISSUE THREAD

**Issue by brainchild0** [whipper-team/whipper#542]
Please excuse me if anyone would prefer   that such a question not be given through this system,  but  I find the specific information nowhere I have looked, nor find any other place to ask.

Is the application currently mature enough that a ripped image  it produces, assuming no error or warning occurred during the process, may be considered reliable, that is, accurate and complete, to the same degree as other from software (e.g. EAC)?

**Comment by JuniorJPDJ** [comment]
Yes, but it's hard to verify it as it only uses AccurateRip to do so and loads of CDs are not here.
EAC and CUERipper also use CTDB, but whipper still doesn't (#15).

**Comment by Bujiraso** [comment]
It's quite reliable here.
Of my 623 CDs, it got 585 of them ripped accurately.

31 of those CDs would not rip because they were not a common pressing. dbpoweramp worked just fine with those. The remainder were damaged or were not in AccurateRip's database, again forcing me to use dbpoweramp to enter my initial values (no certainty of accuracy)

I've yet to take the time to file a bug on that.

**Comment by JoeLametta** [comment]
Whipper should be reliable (not bulletproof, see current limitations):
- multi-sessions CD support is broken (https://github.com/whipper-team/whipper/issues/174)
- Mixed Mode CDs are currently not supported (https://github.com/whipper-team/whipper/issues/170)
- (https://github.com/whipper-team/whipper/issues/88) and (https://github.com/whipper-team/whipper/issues/294)

We also depend both on cdrdao and libcdio's cd-paranoia to perform the ToC/audio extraction so any serious bug affecting one of them is going to cascade issues in whipper (https://github.com/whipper-team/whipper/issues?q=label%3A"Upstream+Bug").

If the CDs you're trying to rip have useful AccurateRip DB records, you can rely on those to give you an idea of the confidence level of the rip you've completed.
In case no online DB entry exists for the CDs you're trying to rip you can always rip it multiple times using different software and compare if the resulting data (audio) matches (Test & Copy with Exact Audio Copy, then whipper, etc.)

**Comment by tsweet64** [comment]
It seems to vary based on hardware. On my drive (reported as a `HL-DT-ST DVDRAM GSA-T10N`) whipper fails in some way with every CD. It either won't read the TOC at all, due to #531 (see my comment there for further details) or if that succeeds it still fails to rip the last track if I set the correct offset (likely upstream rocky/libcdio-paranoia#14).

I've been using abcde instead, but I'd love to have the AccurateRip verification which whipper provides. I'm considering getting a new CD drive to see if it works properly. Does anyone know any good external optical drives that work well with whipper?

**Comment by MerlijnWajer** [comment]
I use the iHAS-124 mostly. But it's not an external optical drive, but it works with most usb<->sata enclosures.
